src/overcooked_ai_py/mdp/overcooked_mdp.py

```diff
@@ -1989,68 +2084,82 @@ def make_closest_feature(idx, name, locations):
                 obj_idx = OBJ_TO_IDX[held_obj_name]
                 all_features["p{}_objs".format(i)] = np.eye(len(IDX_TO_OBJ))[obj_idx]
 
-            # Closest feature of each type
-            if held_obj_name == "onion":
-                all_features["p{}_closest_onion".format(i)] = (0, 0)
-            else:
-                make_closest_feature(i, "onion", self.get_onion_dispenser_locations() + counter_objects["onion"])
-
-            make_closest_feature(i, "empty_pot", pot_state["empty"])
-            make_closest_feature(i, "one_onion_pot", pot_state["1_items"])
-            make_closest_feature(i, "two_onion_pot", pot_state["2_items"])
-            make_closest_feature(i, "cooking_pot", pot_state["cooking"])
-            make_closest_feature(i, "ready_pot", pot_state["ready"])
-
-            if held_obj_name == "dish":
-                all_features["p{}_closest_dish".format(i)] = (0, 0)
-            else:
-                make_closest_feature(i, "dish", self.get_dish_dispenser_locations() + counter_objects["dish"])
-
-            if held_obj_name == "soup":
-                all_features["p{}_closest_soup".format(i)] = (0, 0)
-            else:
-                make_closest_feature(i, "soup", counter_objects["soup"])
-
-            make_closest_feature(i, "serving", self.get_serving_locations())
-
+            # Closest feature for each object type
+            all_features = concat_dicts(all_features, make_closest_feature(i, player, "onion", self.get_onion_dispenser_locations() + counter_objects["onion"]))
+            all_features = concat_dicts(all_features, make_closest_feature(i, player, "tomato", self.get_tomato_dispenser_locations() + counter_objects["tomato"]))
+            all_features = concat_dicts(all_features, make_closest_feature(i, player, "dish", self.get_dish_dispenser_locations() + counter_objects["dish"]))
+            all_features = concat_dicts(all_features, make_closest_feature(i, player, "soup", counter_objects["soup"]))
+            all_features = concat_dicts(all_features, make_closest_feature(i, player, "serving", self.get_serving_locations()))
+            all_features = concat_dicts(all_features, make_closest_feature(i, player, "empty_counter", self.get_empty_counter_locations(overcooked_state)))
+
+            # Closest pots info
+            pot_locations = self.get_pot_locations().copy()
+            for pot_idx in range(num_pots):
+                _, closest_pot_loc = mlam.motion_planner.min_cost_to_feature(player.pos_and_or, pot_locations, with_argmin=True)
+                pot_features = make_pot_feature(i, player, pot_idx, closest_pot_loc, pot_states)
```

Comment:
Just checking that this currently supports the case in which there might be a variable number of pots across different layouts you are interested in, and you want the featurization shape to be the same across those different cases.

For example, if you have two layouts (one with 1 pot and one with 2) and you want to do BC on both layouts simultaneously (rather than training 2 separate agents), you would probably want to set num_pots=2 and then have the 2nd pot slot in the featurization of the first layout have some null token (all 0s or something).

min_cost_to_feature will not return a pot_location, and make_pot_feature will return all zeros?

Reply:
Correct! In that case, `closest_pot_loc` will be `None`, so `make_pot_feature` will put in all zeros for the second pot. Building off of your comment above, I think it's a good idea to add a thorough description of the feature encodings, what exactly `num_pots` does, and how it handles the 'there is no pot to encode' case at the top of the function in a docstring would be worthwhile